### PR TITLE
Fixes #36839 - Optimize apipie-dsl translate method

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Comment by ekohl** [comment]
[test katello]

**Review by ofedoren — DISMISSED** [review]
Thanks, @ekohl, seems right, didn't test though yet.

Should we do the same for apipie-rails translate method?

**Comment by ekohl** [comment]
I didn't test it either, but how about this?

**Review by ofedoren — DISMISSED** [review]
Looks much cleaner, thanks!

AFAIU, `with_locale` is proper (recommended) way instead of reseting with `set_locale`, so functional-wise it should be the same.

File: config/initializers/apipie.rb [hunk 1]
```diff
@@ -19,13 +19,7 @@
...
   config.locale = ->(loc) { loc ? FastGettext.set_locale(loc) : FastGettext.locale }
 
-  config.translate = lambda do |str, loc|
-    old_loc = FastGettext.locale
-    FastGettext.set_locale(loc)
-    trans = _(str) if str
-    FastGettext.set_locale(old_loc)
-    trans
-  end
+  config.translate = ->(stc, loc) { str ? FastGettext.with_locale(loc) { _(str) } : nil }
```
Comment (ofedoren) [inline config/initializers/apipie.rb]:
```suggestion
  config.translate = ->(str, loc) { str ? FastGettext.with_locale(loc) { _(str) } : nil }
```

Comment (ekohl) [reply config/initializers/apipie.rb]: Argh, I had that fixed on my laptop, then rebased on my desktop where I made the mistake in the first place ...

**Comment by ofedoren** [comment]
I don't think Katello failures are related, but just in case...

[test katello]

**Review by ofedoren — APPROVED** [review]
Thanks, @ekohl, merging.

**Pull request merged**